gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/mfa/MFAEnrollEndpoint.java

```diff
@@ -173,20 +171,28 @@ private List<Factor> factorsToRender(List<Factor> factors, Session session, Clie
         // if an alternative factor ID has been set, only display this one
         final String alternativeFactorId = session.get(ConstantKeys.ALTERNATIVE_FACTOR_ID_KEY);
         var mfaContext = new MfaFilterContext(context, client, factorManager, ruleEngine);
-        var applicationFactors = ofNullable(client.getFactorSettings()).orElseGet(FactorSettings::new);
         if (alternativeFactorId != null && !alternativeFactorId.isEmpty()) {
             // check if this alternative factor still exists
             Optional<Factor> optionalFactor = factors.stream()
-                    .filter(factor -> alternativeFactorId.equals(factor.getId()) && mfaContext.matchFactorRule(factor.getId(), applicationFactors))
+                    .filter(factor -> alternativeFactorId.equals(factor.getId()))
```

Comment:
why "&& mfaContext.matchFactorRule(factor.getId(), applicationFactors))" is removed?